connectivity/tests/endpointslice-clustermesh-sync.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package tests
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/utils/features"
+)
+
+func EndpointSliceClusterMeshSync() check.Scenario {
+	return &endpointSliceClusterMeshSync{}
+}
+
+type endpointSliceClusterMeshSync struct{}
+
+func (s *endpointSliceClusterMeshSync) Name() string {
+	return "endpointslice-clustermesh-sync"
```

Comment:
I've personally a small preference towards `clustermesh-endpointslice-sync` (same for the variable names), but no strong feelings. Feel free to keep it as it is now.